internal/persistence/sql/relationtuples.go

```diff
@@ -260,15 +272,63 @@ func (p *Persister) ExistsRelationTuples(ctx context.Context, query *relationtup
 	return exists, sqlcon.HandleError(err)
 }
 
+func buildInsert(commitTime time.Time, nid uuid.UUID, rs []*relationtuple.RelationTuple) (query string, args []any, err error) {
+	if len(rs) == 0 {
+		return "", nil, errors.WithStack(ketoapi.ErrMalformedInput)
+	}
+
+	var q strings.Builder
+	fmt.Fprintf(&q, "INSERT INTO %s (shard_id, nid, namespace, object, relation, subject_id, subject_set_namespace, subject_set_object, subject_set_relation, commit_time) VALUES ", (&RelationTuple{}).TableName())
```

Comment:
As the nid and commit time are always the same, we could do this optimized query instead (simplified example):
```sql
INSERT INTO vals (nid, id, s)
SELECT 
    $1 AS nid, 
    gen_random_uuid() AS id, 
    v.s
FROM 
    (VALUES ($2), ($3), ($3)) AS v(s);
```
Not a blocker IMO, I'm happy either way.

Reply:
I don't think it is valid SQL to have `SELECT $1`. Either way, the chunk size is IMO large enough.